cep-recipe-serialization.md

```diff
@@ -0,0 +1,1117 @@
+> and fill out example for `pin_compatible`. Should remove double
+> `pin_subpackage` in rattler-build.
+
+- Pin expressions like ${{ pin_subpackage('libtool', max_pin='x.x') }} are
+  rendered to:
+
+  ```yaml
+  - ${{ pin_subpackage('libtool', exact=true) }}
+  # becomes
+  - pin_subpackage:
+      pin_subpackage:
+        name: libtool
+        max_pin: null
+        min_pin: null
+        exact: true
+  ```
+
+- For the following:
+
+  ```yaml
+  - ${{ pin_subpackage('some-pkg-a', min_pin="x.x.x") }}
+  - ${{ pin_compatible('python', min_pin="x.x.x") }}
+  ```
+
+  the rendered result is as follows:
+
+  ```yaml
+  - pin_subpackage:
+      pin_subpackage:
+        name: some-pkg-a
+        max_pin: null
+        min_pin: x.x.x
+        exact: false
+  - pin_compatible:
+      pin_compatible:
+        name: python
+        max_pin: null
+        min_pin: x.x.x
+        exact: false
+  ```
+
+### Build configuration section
+
+This records all information that was used at build time, incl. variables given
+on the CLI or by environment yaml. Everything necessary to reproduce the build
+should be recorded here.
+
+Here's the documentation for the fields in the provided YAML:
+
+#### `build_configuration`
+
+The `build_configuration` section contains information related to the build
+process.
+
+##### `target_platform`
+
+- Type: string
+- Description: The target platform for which the package is being built (e.g.,
+  `osx-arm64`).
+
+##### `host_platform`
+
+- Type: string
+- Description: The host platform used for building the package. Usually
+  equivalent to `target_platform`, except when the `target_platform` is
+  `noarch`. If the `target_platform` is `noarch`, the `host_platform` is the
+  platform where the build is being executed.
+
+##### `build_platform`
+
+- Type: string
+- Description: The platform used for building the package (e.g., `osx-arm64`).
+
+##### `variant`
+
+- Type: object
+- Description: Dictionary with the variant configuration used for the build.
+
+##### `hash`
+
+- Type: object
+- Description: Contains information related to hashing.
+  - `hash`: The hash value (e.g., `60d57d3`).
+  - `hash_input`: The input values used for generating the hash.
+  - `hash_prefix`: Prefix for the hash, which can include values like `py311`
+    based on the variant configuration.
+
+##### `directories`
+
+- Type: object
+- Description: Contains directory paths used during the build process.
+  - `host_prefix`: The host prefix directory with placeholders for the host
+    environment.
+  - `build_prefix`: The build prefix directory for the build environment.
+  - `work_dir`: The working directory with extracted sources.
+  - `build_dir`: The build directory that contains `host_prefix`,
+    `build_prefix`, and `work_dir` as subdirectories.
+
+##### `channels`
+
+- Type: array of strings
+- Description: The conda channels used for resolving dependencies (e.g.,
```

Comment:
Is this expecting the name of the channel? That might be ambiguous. What about the resolved URL? I guess without authentication details. Also, given how the resolved dependencies already contain the lockfile-like info... is this needed?

Reply:
If we want to fully specify this, I think it's more specific to specify which repodata URLs where used and in which order and priority style (strict, flexible, etc).

Reply:
This should be the resolved URLs indeed.